Src/FluentAssertions/Collections/GenericCollectionAssertions.cs

```diff
@@ -43,7 +43,7 @@ public AndConstraint<GenericCollectionAssertions<T>> NotContainNulls<TKey>(Expre
 
             Func<T, TKey> compiledPredicate = predicate.Compile();
 
-            var values = Subject
+            T[] values = Subject
```

Comment:
According AV1520 the type is clear enough.

Reply:
Knowing that it's an array is valuable info.

Reply:
The line ends with `ToArray`. Isn't clear enough?